File: android/app/src/androidTest/java/com/bitbusters/android/speproject/SplashActivityTest.java
package com.bitbusters.android.speproject;

import android.content.Context;
import android.content.Intent;
import android.support.test.filters.LargeTest;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;
import android.support.test.uiautomator.By;
import android.support.test.uiautomator.UiDevice;
import android.support.test.uiautomator.UiObject;
import android.support.test.uiautomator.UiSelector;
import android.support.test.uiautomator.Until;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.InstrumentationRegistry.getContext;
import static android.support.test.InstrumentationRegistry.getInstrumentation;
import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static junit.framework.Assert.assertTrue;
import static org.hamcrest.core.IsNot.not;
import static org.hamcrest.core.IsNull.notNullValue;
import static org.junit.Assert.assertThat;

/**
 * Created by mihajlo on 12/06/17.
 */

@RunWith(AndroidJUnit4.class)
@LargeTest
public class SplashActivityTest extends TestHelper {

    private static final String TAG = "DATA_VIEW_ACTIVITY_TEST";

    private static final String FOLDER_NAME = "SplashActivityTest";

    private static final String PACKAGE = "com.bitbusters.android.speproject";

    private static final int LAUNCH_TIMEOUT = 5000;

    private UiDevice mDevice;

    @Rule
    public ActivityTestRule<DataViewActivity> mActivityRule = new ActivityTestRule<>(DataViewActivity.class);

    @Before
    public void setActivity() {
        // Start the activity
        mActivityRule.getActivity();
        // Access the device state using UiAutomator
        mDevice = UiDevice.getInstance(getInstrumentation());
        // Start from the home screen
        mDevice.pressHome();
    }

    @Test
    public void SplashActivityTestLaunchApp() {
        // Wait for launcher
        final String launcherPackage = mDevice.getLauncherPackageName();
        assertThat(launcherPackage, notNullValue());
        mDevice.wait(Until.hasObject(By.pkg(launcherPackage).depth(0)),
                LAUNCH_TIMEOUT);

        // Launch the app
        Context context = getContext();
        final Intent intent = context.getPackageManager().getLaunchIntentForPackage(PACKAGE);

        // Clear out any previous instances
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
        context.startActivity(intent);

        // Wait for the app to appear
        mDevice.wait(Until.hasObject(By.pkg(PACKAGE).depth(0)), LAUNCH_TIMEOUT);

        takeScreenshot(mDevice, FOLDER_NAME, "SplashActivityTestLaunchApp");

        onView(withId(R.id.map)).check(matches(isDisplayed()));
        onView(withId(R.id.gps_button)).check(matches(isDisplayed()));
    }
}

File: android/app/src/androidTest/java/com/epimorphics/android/myrivers/SplashActivityTest.kt
package com.epimorphics.android.myrivers

import android.content.Intent
import android.support.test.filters.LargeTest
import android.support.test.rule.ActivityTestRule
import android.support.test.runner.AndroidJUnit4
import android.support.test.uiautomator.By
import android.support.test.uiautomator.UiDevice
import android.support.test.uiautomator.Until

import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

import android.support.test.InstrumentationRegistry.getContext
import android.support.test.InstrumentationRegistry.getInstrumentation
import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.assertion.ViewAssertions.matches
import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
import android.support.test.espresso.matcher.ViewMatchers.withId


/**
 * Created by mihajlo on 28/07/17.
 */

@RunWith(AndroidJUnit4::class)
@LargeTest
class SplashActivityTest: TestHelper() {

    private val FOLDER_NAME = "SplashActivityTest"

    private val PACKAGE = "com.epimorphics.android.myrivers"

    private val LAUNCH_TIMEOUT = 5000

    private lateinit var mDevice: UiDevice

    @Rule @JvmField
    var mActivityRule = ActivityTestRule(DataViewActivity::class.java)

    @Before
    fun setActivity() {
        // Start the activity
        mActivityRule.activity
        // Access the device state using UiAutomator
        mDevice = UiDevice.getInstance(getInstrumentation())
        // Start from the home screen
        mDevice.pressHome()
    }

    @Test
    fun SplashActivityTestLaunchApp() {
        // Wait for launcher
        val launcherPackage = mDevice.launcherPackageName
        mDevice.wait(Until.hasObject(By.pkg(launcherPackage).depth(0)), LAUNCH_TIMEOUT.toLong())

        // Launch the app
        val context = getContext()
        val intent = context.packageManager.getLaunchIntentForPackage(PACKAGE)

        // Clear out any previous instances
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
        context.startActivity(intent)

        // Wait for the app to appear
        mDevice.wait(Until.hasObject(By.pkg(PACKAGE).depth(0)), LAUNCH_TIMEOUT.toLong())

        takeScreenshot(mDevice, FOLDER_NAME, "SplashActivityTestLaunchApp")

        onView(withId(R.id.map)).check(matches(isDisplayed()))
        onView(withId(R.id.gps_button)).check(matches(isDisplayed()))
    }
}